src/libstd/ffi/c_str.rs

```diff
@@ -1040,8 +1040,8 @@ impl CStr {
     /// ```
     #[inline]
     #[stable(feature = "cstr_from_bytes", since = "1.10.0")]
-    pub unsafe fn from_bytes_with_nul_unchecked(bytes: &[u8]) -> &CStr {
-        &*(bytes as *const [u8] as *const CStr)
```

Comment:
Oh. I should have checked the original source :D You should be able to just add `#![feature(const_raw_ptr_deref)]` and keep the original source ;)